File: gmao_mobile/lib/models/batiment_maintenance_stats.dart
class BatimentMaintenanceStats {
  final int batimentId;
  final String batimentName;
  final int activeInterventionCount;
  final int criticalPriorityCount;
  final String status; // SAFE, WARNING, CRITICAL

  BatimentMaintenanceStats({
    required this.batimentId,
    required this.batimentName,
    required this.activeInterventionCount,
    required this.criticalPriorityCount,
    required this.status,
  });

  factory BatimentMaintenanceStats.fromJson(Map<String, dynamic> json) {
    return BatimentMaintenanceStats(
      batimentId: json['batimentId'],
      batimentName: json['batimentName'] ?? 'Inconnu',
      activeInterventionCount: json['activeInterventionCount'] ?? 0,
      criticalPriorityCount: json['criticalPriorityCount'] ?? 0,
      status: json['status'] ?? 'SAFE',
    );
  }
}

File: gmao_backend/src/main/java/com/gmao/backend/dto/BatimentMaintenanceStats.java
package com.gmao.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BatimentMaintenanceStats {
    private Long batimentId;
    private String batimentName;
    private long activeInterventionCount;
    private long criticalPriorityCount;
    private String status; // SAFE, WARNING, CRITICAL
}
